src/olympia/core/utils.py

```diff
@@ -5,15 +5,27 @@
 import django
 
 
-def get_version_json():
-    contents = {}
+# Keys required to be set in the version.json file.
+REQUIRED_VERSION_KEYS = ['target', 'version', 'source', 'commit', 'build']
 
-    root = os.path.join(os.path.dirname(os.path.dirname(__file__)), '..', '..')
+root = os.path.join(os.path.dirname(os.path.dirname(__file__)), '..', '..')
+pkg_json_path = os.path.join(root, 'package.json')
 
-    contents['commit'] = os.environ.get('DOCKER_COMMIT', 'commit')
-    contents['version'] = os.environ.get('DOCKER_VERSION', 'local')
-    contents['source'] = 'https://github.com/mozilla/addons-server'
-    contents['build'] = os.environ.get('DOCKER_BUILD', 'build')
+
+def get_version_json(
+    build_info_path=os.environ['BUILD_INFO'],
+    pkg_json_path=pkg_json_path,
+):
+    contents = {key: '' for key in REQUIRED_VERSION_KEYS}
+
+    # Read the build info from the docker image.
+    # This is static read only data that cannot
+    # be overridden at runtime.
+    if os.path.exists(build_info_path):
+        with open(build_info_path) as f:
+            data = json.loads(f.read())
+            for key, value in data.items():
+                contents[key] = value
```

Comment:
nit: you could probably replace this with something like:
```suggestion
            contents.update(json.loads(f.read()))
```
?